src/plugins/stake-plugins/generic/policyAdapters/CardanoKilnAdaptor.ts

```diff
@@ -71,7 +73,21 @@ export const makeCardanoKilnAdapter = (policyConfig: StakePolicyConfig<CardanoPo
         throw new Error('Insufficient funds')
       }
 
-      const stakeTransaction = await kiln.adaStakeTransaction(walletAddress, adapterConfig.poolId, accountId)
+      const result = await kiln.adaStakeTransaction(walletAddress, adapterConfig.poolId, accountId).catch(error => {
+        if (error instanceof Error) return error
+        throw error
+      })
+      if (result instanceof KilnError) {
```

Comment:
The KilnError class seems unnecessary since you don't need the additional .error string. You could just use an Error and .message

Reply:
Not necessarily. I don't know what Kiln will throw at us in the future. By modeling the error after their error response, we can handle anything they throw at us.